sys/riscv/riscv/elf_machdep.c

```diff
@@ -492,6 +500,21 @@ elf_reloc_internal(linker_file_t lf, Elf_Addr relocbase, const void *data,
 			    before32, *insn32p);
 		break;
 
+#ifdef __CHERI_PURE_CAPABILITY__
+	case R_RISCV_CHERI_CAPABILITY:
+		error = LINKER_SYMIDX_CAPABILITY(lf, symidx, 1, &cap);
+		if (error != 0)
+			return (-1);
+
+		cap += addend;
+		beforecap = *(uintcap_t *)where;
+		*(uintcap_t *)where = cap;
+		if (debug_kld)
+			printf("%p %c %-24s %#p -> %#p\n", where,
+			    (local ? 'l' : 'g'), reloctype_to_str(rtype),
+			    (void *)beforecap, (void *)cap);
```

Comment:
`%#lp` and `void * __capability`?

Reply:
Only purecap kernels do this, so there's no need for the `__capability`.  Perhaps it could use `%#p` though.

Reply:
Hybrid kernels will eventually want to, but the reason I commented is because it's using `uintcap_t` too, so mixing implicit and explicit.

Reply:
Adapted so that we can switch it on for hybrid kernels as well if we want to.